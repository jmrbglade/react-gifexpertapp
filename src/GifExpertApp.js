import React, { useState } from 'react';
// import PropTypes from 'prop-types';
import { AddCategory } from './components/AddCategory';
import GifGrid from './components/GifGrid'; 

const GifExpertApp = ({ defaultCategories = [] }) => {

    // const categories = ['One Punch', 'Samurai Jack', 'Dragon Ball'];
    const [categories, setCategories] = useState( defaultCategories );
    
    // const handleAdd = () => {
    //     setCategories([...categories, 'Yakusoku no Neverland']);
    // }

    return (
        <>
          <h2>GifExpertApp</h2>
          <AddCategory setCategories={ setCategories }/>
          <hr /> 
          {/* <button onClick={ handleAdd }>Agregar</button> */}
          <div>
              {
                categories.map( category => (
                    <GifGrid 
                        key={ category }
                        category={ category } 
                    />
                    // <li key={ category }>{ category }</li>
                ))
              }
          </div>
        </>
    );
}

// GifExpertApp.propTypes = {

// }

export default GifExpertApp;
