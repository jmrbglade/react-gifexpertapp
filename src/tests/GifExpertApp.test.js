import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";
import React from 'react';

describe('Pruebas en <GifExpertApp />', () => {
    test('Debe mostrar el componente correctamente', () => {
        const wrapper = shallow( <GifExpertApp />);
        expect( wrapper ).toMatchSnapshot();
    })
    test('Debe mostrar una lista de categorÃ­as', () => {
        const categories = ['One Punch Man', 'Samurai Jack'];
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories }/>);
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    })
    
})
