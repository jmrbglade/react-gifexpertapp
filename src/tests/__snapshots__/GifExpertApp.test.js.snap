// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <div />
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> Debe mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <div>
    <GifGrid
      category="One Punch Man"
      key="One Punch Man"
    />
    <GifGrid
      category="Samurai Jack"
      key="Samurai Jack"
    />
  </div>
</Fragment>
`;
